django/db/backends/postgresql/operations.py

```diff
@@ -237,20 +251,18 @@ def tablespace_sql(self, tablespace, inline=False):
             return "TABLESPACE %s" % self.quote_name(tablespace)
 
     def sequence_reset_sql(self, style, model_list):
-        from django.db import models
-
         output = []
         qn = self.quote_name
+
         for model in model_list:
             # Use `coalesce` to set the sequence for each model to the max pk
             # value if there are records, or 1 if there are none. Set the
             # `is_called` property (the third argument to `setval`) to true if
             # there are records (as the max pk value is already in use),
             # otherwise set it to false. Use pg_get_serial_sequence to get the
             # underlying sequence name from the table name and column name.
-
             for f in model._meta.local_fields:
-                if isinstance(f, models.AutoField):
+                if isinstance(f, (AutoFieldMixin, SerialFieldMixin)):
```

Comment:
Small aside: This is an interesting one. There's a wider issue around replacing `isinstance(..., AutoField)` with something else - most likely some kind of `is_generated` or ` is_auto` or something - using a flag, rather than `isinstance()`. I'd have to look up the details as it was a number of years ago. It's probably not going to go in the direction of using `isinstance(..., AutoFieldMixin)` though... 🤔

Reply:
Yeah, having a flag for this would make this code more extensible. Does it have to be addressed in this PR though?
By the way, as far as I understand, this function is broken for `BigAutoField` and `SmallAutoField`, that's why I replaced it with the mixin.

Reply:
> Does it have to be addressed in this PR though?

No, was just pondering whether we wanted to do this isinstance of mixin things. I guess it's internal, so it's fine.

Reply:
I'm finally catching up on this PR; @ngnpope are you referring to the `uses_sequence` flag from this old PR for the same issue?
https://github.com/django/django/pull/12989/files#diff-ef0fe18fcc05a65a19d03870445490ac8a49a12bcfe3c1f8b6879d7b9cdc1d82R177